Backpack/FlareView/Classes/BPKFlareView.m

```diff
@@ -84,23 +84,53 @@ - (void)setup {
     [self addSubview:self.contentView];
     self.contentView.translatesAutoresizingMaskIntoConstraints = NO;
 
-    self.contentViewBottomConstraint = [self.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor
-                                                                         constant:BPKFlareHeight];
-
     [NSLayoutConstraint activateConstraints:@[
         [self.backgroundView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
         [self.backgroundView.topAnchor constraintEqualToAnchor:self.topAnchor],
         [self.trailingAnchor constraintEqualToAnchor:self.backgroundView.trailingAnchor],
         [self.bottomAnchor constraintEqualToAnchor:self.backgroundView.bottomAnchor],
     ]];
 
-    [NSLayoutConstraint activateConstraints:@[
+    [self updateContentViewConstraints];
+    [self createLayerMask];
+}
+
+- (void)updateContentViewConstraints {
+    if (self.contentViewConstraints != nil) {
+        [NSLayoutConstraint deactivateConstraints:self.contentViewConstraints];
+    }
+
+    CGFloat topConstant = 0;
+    CGFloat bottomConstant = 0;
+    switch (self.flarePosition) {
+        case BPKFlarePositionBottom:
+            topConstant = 0;
+            bottomConstant = BPKFlareHeight;
+            break;
+        case BPKFlarePositionTop:
+            topConstant = BPKFlareHeight;
+            bottomConstant = 0;
+            break;
+    }
+
+    self.contentViewConstraints = @[
         [self.contentView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
-        [self.contentView.topAnchor constraintEqualToAnchor:self.topAnchor],
-        [self.trailingAnchor constraintEqualToAnchor:self.contentView.trailingAnchor], self.contentViewBottomConstraint
-    ]];
+        [self.contentView.topAnchor constraintEqualToAnchor:self.topAnchor constant:topConstant],
+        [self.trailingAnchor constraintEqualToAnchor:self.contentView.trailingAnchor],
+        [self.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor constant:bottomConstant]
+    ];
+    [NSLayoutConstraint activateConstraints:self.contentViewConstraints];
+}
 
-    [self createLayerMask];
+- (void)setFlarePosition:(BPKFlarePosition)flarePosition {
+    if (_flarePosition != flarePosition) {
+        _flarePosition = flarePosition;
+        [self updateContentViewConstraints];
+        [self createLayerMask];
+
+        [self setNeedsLayout];
+        [self layoutIfNeeded];
```

Comment:
This shouldn't be required `setNeedsLayout` already causes the view to be laid out in the next layout pass.

Reply:
This was needed to make the test work, but the example in the app seemed to be fine without it. Maybe I should move it to the test method only?

Reply:
Yeah that sounds sensible